### PR TITLE
[DBMON-2461] Bump up go-sqllexer to support collecting procedure names

### PR DESCRIPTION
**Pull request by lu-zhengda**
### What does this PR do?
This PR bumps `go-sqllexer` to v0.0.7 to support collecting stored procedure names. 

### Motivation
The stored procedure names were used to be collected in Python integrations and the logic requires (re)traverse the entire sql statement. By moving the procedure name collection to the normalization process not only reduces unnecessary rescanning of the sql statement, but also enriches the collected sql metadata.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by alexandre-normand — APPROVED** [review]

File: releasenotes/notes/go-sqllexer-minor-version-bump-b81be45812e4a840.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Bump the minimum version of the `go-sqllexer` library to 0.0.7 to support collect stored procedure names.
```
Comment (hestonhoffman) [inline releasenotes/notes/go-sqllexer-minor-version-bump-b81be45812e4a840.yaml:11]:
```suggestion
    Bump the minimum version of the `go-sqllexer` library to 0.0.7 to support collecting stored procedure names.
```

**Review by hestonhoffman — APPROVED** [review]
small edit

File: releasenotes/notes/go-sqllexer-minor-version-bump-b81be45812e4a840.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Bump the minimum version of the `go-sqllexer` library to 0.0.7 to support collecting stored procedure names.
```
Comment (darccio) [inline releasenotes/notes/go-sqllexer-minor-version-bump-b81be45812e4a840.yaml:11]:
```suggestion
    DBM: Bump the minimum version of the `go-sqllexer` library to 0.0.7 to support collecting stored procedure names.
```

**Review by darccio — APPROVED** [review]

**Review by remeh — APPROVED** [review]
LGTM for AML-owned files 👍

**Pull request merged**